lib/src/lib.rs

```diff
@@ -65,6 +65,7 @@ struct Parameters {
     output_size: (u32, u32),
     guide_alpha: f32,
     random_resolve: Option<u64>,
+    max_thread_count: Option<u32>,
```

Comment:
Minor nitpick: technically as the # threads is a count it probably should be `usize` type. Though I doubt one would use more than 4 million threads, but you don't have to cast then in the map on usage.

```suggestion
    max_thread_count: Option<usize>,
```